maven-embedder/src/main/java/org/apache/maven/cli/transfer/AbstractMavenTransferListener.java

```diff
@@ -214,18 +227,38 @@ public void transferSucceeded(TransferEvent event) {
         FileSizeFormat format = new FileSizeFormat(Locale.ENGLISH);
 
         StringBuilder message = new StringBuilder();
-        message.append(action).append(' ').append(direction).append(' ').append(resource.getRepositoryId());
-        message.append(": ");
-        message.append(resource.getRepositoryUrl()).append(resource.getResourceName());
-        message.append(" (").append(format.format(contentLength));
+        message.append(action).append(darkOn).append(' ').append(direction).append(' ');
+        message.append(darkOff).append(resource.getRepositoryId());
+        message.append(darkOn).append(": ");
+        message.append(resource.getRepositoryUrl()).append(darkOff);
+        appendResource(message, resource.getResourceName());
+        message.append(darkOn).append(" (").append(format.format(contentLength));
 
         long duration = System.currentTimeMillis() - resource.getTransferStartTime();
         if (duration > 0L) {
             double bytesPerSecond = contentLength / (duration / 1000.0);
             message.append(" at ").append(format.format((long) bytesPerSecond)).append("/s");
         }
 
-        message.append(')');
+        message.append(')').append(darkOff);
         out.println(message.toString());
     }
+
+    private void appendResource(StringBuilder message, String resource) {
+        if (!MessageUtils.isColorEnabled() || resource.endsWith("/maven-metadata.xml")) {
+            message.append(resource);
+            return;
+        }
+        // use Maven2 layout for non-metadata files: https://maven.apache.org/repositories/layout.html
```

Comment:
Please don't do this: we want a LONG road from Maven2 to NOT have code (and assumptions) like this sprinkled across the codebase.

The API offers you repositoryBaseUrl and relative resource path, so just color those two differently (as I agree, repo URL IS important, it may be affected by user settings) while resource path IS NOT important....

Reply:
By the way, since Maven3 it is called "default" layout, not "maven2 layout" (as Maven2 did support "maven1" layout as well, but Maven3 does not).

Reply:
please don't do what? use Maven 2 repository layout as a fact?

Reply:
Please do not make wrong assumptions.